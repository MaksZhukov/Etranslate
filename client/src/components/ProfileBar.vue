<template>
  <v-menu offset-y>
    <v-btn slot="activator" flat>
      {{email}}
      <v-icon>more_vert</v-icon>
    </v-btn>
    <v-list>
      <v-list-tile
        :to="{ name: 'dashboard', query: textLang.abbr && translateLang && inputText ? { text: inputText,
        textLang: textLang.abbr, translateLang: translateLang.abbr } : {}}"
      >Dashboard</v-list-tile>
      <v-list-tile to="/dictionary">Dictionary</v-list-tile>
      <v-list-tile @click="logOut">
        <v-list-tile-title>Logout</v-list-tile-title>
      </v-list-tile>
    </v-list>
  </v-menu>
</template>


<script>

import { mapMutations, mapState } from 'vuex';

export default {
  name: 'profileBar',
  props: {
    email: {
      type: String,
      required: true,
    },
  },
  computed: {
    ...mapState('translateModule', ['inputText', 'textLang', 'translateLang']),
  },
  methods: {
    ...mapMutations('userModule', ['logOut']),
  },
};
</script>
