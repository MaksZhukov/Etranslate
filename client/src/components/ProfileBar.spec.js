import { mount, createLocalVue, RouterLinkStub } from '@vue/test-utils';
import Vuetify from 'vuetify';
import Vuex from 'vuex';
import Fragment from 'vue-fragment';
import VueRouter from 'vue-router';
import ProfileBar from './ProfileBar.vue';


describe('ProfileBar component', () => {
  let wrapper;
  let mutations;
  const email = 'example@mail.ru';
  beforeEach(() => {
    const localVue = createLocalVue();
    localVue.use(Vuetify);
    localVue.use(Vuex);
    localVue.use(Fragment.Plugin);
    localVue.use(new VueRouter());
    mutations = {
      logOut: jest.fn(),
    };
    const store = new Vuex.Store({
      modules: {
        userModule: {
          namespaced: true,
          state: {},
          mutations,
        },
      },
    });
    wrapper = mount(ProfileBar, {
      localVue, store, propsData: { email }, stubs: { RouterLink: RouterLinkStub },
    });
  });
  it('profilebar shapshots have different email', () => {
    expect(wrapper).toMatchSnapshot();
    wrapper.setProps({ email: 'newexample.@mail.ru' });
    expect(wrapper).toMatchSnapshot();
  });
  it(`profilebar email is ${email}`, () => {
    expect(wrapper.props('email')).toBe(email);
    expect(wrapper.find('.v-menu').text()).toContain(email);
  });
  it('profilebar has logout and dashboard link', () => {
    const arrayA = wrapper.findAll('a');
    expect(arrayA.at(1).text()).toContain('Dashboard');
    expect(arrayA.at(2).text()).toContain('Logout');
  });
  it('profilebar click logout was execute', () => {
    const arrayA = wrapper.findAll('a');
    arrayA.at(2).trigger('click');
    expect(mutations.logOut).toHaveBeenCalled();
  });
});
