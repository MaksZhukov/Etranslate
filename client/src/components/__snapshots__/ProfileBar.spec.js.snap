// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileBar component profilebar shapshots have different email 1`] = `
<div class="v-menu v-menu--inline">
  <div class="v-menu__activator"><button type="button" class="v-btn v-btn--flat theme--light">
      <div class="v-btn__content">
        example@mail.ru
        <i aria-hidden="true" class="v-icon material-icons theme--light">more_vert</i></div>
    </button></div>
</div>
`;

exports[`ProfileBar component profilebar shapshots have different email 2`] = `
<div class="v-menu v-menu--inline">
  <div class="v-menu__activator"><button type="button" class="v-btn v-btn--flat theme--light">
      <div class="v-btn__content">
        newexample.@mail.ru
        <i aria-hidden="true" class="v-icon material-icons theme--light">more_vert</i></div>
    </button></div>
</div>
`;
